File: GameService.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({ providedIn: 'root' })
export class GameService {
  private api = 'http://localhost:8080/api/2048';
  constructor(private http: HttpClient) {}

  getBoard() { return this.http.get<any>(`${this.api}/board`); }
  move(direction: string) { return this.http.post<any>(`${this.api}/move/${direction}`, {}); }
  restart() { return this.http.post(`${this.api}/restart`, {}); }
}

File: GameService.java


import com.example.game2048.model.TileBoard;
import com.example.game2048.model.Direction;
import org.springframework.stereotype.Service;

@Service
public class GameService {
    private TileBoard board = new TileBoard(4);

    public TileBoard getBoard() { return board; }
    public void restart() { board = new TileBoard(4); }
    public TileBoard move(Direction d) { board.move(d); return board; }
}
